### PR TITLE
File uploader

### PR DESCRIPTION
**Pull request by chrisolsen**

### CODE REVIEW

**Comment by Radha-Alberta** [comment]
- The highlighted text says March 21th instead of March 21st. And the time uploaded is shown as 12:58 AM instead of 12:58 PM

image

- The file upload button in figma is of type secondary, but the button in storybook is of type primary

image
image

- The file icons for PDF and image is shown correctly as in Figma. For other file types, it is showing generic icon
image

**Comment by chrisolsen** [comment]
## Fixes

Screenshot_20230322_105046

File: libs/web-components/src/common/styling.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
-export type Spacing = "none" | "3xs" | "2xs" | "xs" | "s" | "m" | "l" | "xl" | "2xl" | "3xl" | "4xl" | string;
+export type Spacing = 
+  "none" | "3xs" | "2xs" | "xs" | "s" | "m" | "l" | "xl" | "2xl" | "3xl" | "4xl" |
+  "0" | "1" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9";
```
Comment (vanessatran-ddi) [inline libs/web-components/src/common/styling.ts:3]: Question: Is it a redundant "1" here?

File: libs/web-components/src/components/icon/Icon.svelte [hunk 2]
```diff
@@ -614,14 +612,37 @@
     color: var(--hover-color);
   }
 
+  .goa-icon--small {
+    width: 1.25rem;
```
Comment (vanessatran-ddi) [inline libs/web-components/src/components/icon/Icon.svelte:616]: Suggestion: We can use design token --goa-icon-size-m

File: libs/web-components/src/components/file-upload-card/FileUploadCard.svelte [hunk 3]
```diff
@@ -0,0 +1,230 @@
+<svelte:options tag="goa-file-upload-card" />
```
Comment (vanessatran-ddi) [inline libs/web-components/src/components/file-upload-card/FileUploadCard.svelte:1]: Question: Should we document this for VSCode json file?

**Review by vanessatran-ddi — COMMENTED** [review]
It is a very huge work. I have only 3 comments. Thank you Chris!

File: libs/web-components/src/components/form-stepper/FormStepper.svelte [hunk 4]
```diff
@@ -147,10 +147,10 @@
 
   /* 2.5rem = 40px where 40px = height of form stepper icons */
   /* 2px = 4px / 2, where 4px is thickness of progress bar */
-  /* 1.75rem = 28px = padding as per design specs */
+  /* 1.5rem = 24px = padding as per design specs */
```
Comment (ArakTaiRoth) [inline libs/web-components/src/components/form-stepper/FormStepper.svelte:150]: Why the changes in this file as well?

File: libs/web-components/src/components/icon/Icon.svelte (hunk 2 above)
Comment (chrisolsen) [reply libs/web-components/src/components/icon/Icon.svelte:616]: The design token values don't seem to line up to the icon named values so I will leave this out until this is resolved  ex .goa-icon--small: var(--goa-icon-size-m)  (small / m)

File: libs/docs/src/components/common/FileUpload.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,359 @@
...
+  {UploadTemplate.bind({})}
+</Story>
+
+<Tabs>
+  <Tab label="Angular">
+    <CodeSnippet
+      lang="ts"
+      code={`
+      export class FileUploadComponent {
+        uploads: Upload[] = [];
```
[the remark this answers is not in the document]
Comment (chrisolsen) [reply libs/docs/src/components/common/FileUpload.stories.mdx:174]: I think *Mock*Uploader is fine.

File: libs/web-components/src/components/button/Button.svelte [hunk 6]
```diff
@@ -255,18 +255,13 @@
 
   .tertiary.destructive {
     color: var(--goa-color-emergency-default);
-    border-color: var(--goa-color-greyscale-200);
```
[the remark this answers is not in the document]
Comment (chrisolsen) [reply libs/web-components/src/components/button/Button.svelte:258]: It's in it's own commit

File: libs/web-components/src/components/form-stepper/FormStepper.svelte (hunk 4 above)
Comment (chrisolsen) [reply libs/web-components/src/components/form-stepper/FormStepper.svelte:150]: it's in it's own commit

File: libs/docs/src/components/common/FileUpload.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,359 @@
...
+        const [uploads, setUploads] = useState<Upload[]>([]);
+        const [progressList, setProgressList] = useState<Record<string, number>>({});
+        function uploadFile(file: File) {
+          const reader = new FileReader()
+          reader.onload = (e: ProgressEvent<FileReader>) => {
+            if (!e.target || !e.target.result) return;
+            if (uploads.find(upload => upload.file.name === file.name)) {
+              return;
+            }
+            const url = e.target.result;
```
[the remark this answers is not in the document]
Comment (chrisolsen) [reply libs/docs/src/components/common/FileUpload.stories.mdx:234]: Both checks are logically valid. The check in the code snippet is probably the better version. I have changed the working code to it.

File: libs/web-components/src/components/file-upload-card/FileUploadCard.svelte (hunk 3 above)
Comment (ArakTaiRoth) [reply libs/web-components/src/components/file-upload-card/FileUploadCard.svelte:1]: @chrisolsen You're still missing the `doc.md` files

File: libs/web-components/src/components/button/Button.svelte (hunk 6 above)
[the remark this answers is not in the document]
Comment (ArakTaiRoth) [reply libs/web-components/src/components/button/Button.svelte:258]: @chrisolsen That doesn't answer the question. I'm wondering why are we making changes to `Button`? Is this a story somewhere to make these changes, were they requested by someone?

File: libs/web-components/src/components/form-stepper/FormStepper.svelte (hunk 4 above)
Comment (ArakTaiRoth) [reply libs/web-components/src/components/form-stepper/FormStepper.svelte:150]: @chrisolsen That doesn't answer the question. I'm wondering why are we making changes to FormStepper? Is this a story somewhere to make these changes, were they requested by someone?

**Comment by chrisolsen** [comment]
@ArakTaiRoth jokes are baaaaad :)

File: libs/web-components/src/components/file-upload-card/FileUploadCard.svelte (hunk 3 above)
Comment (ArakTaiRoth) [reply libs/web-components/src/components/file-upload-card/FileUploadCard.svelte:1]: @chrisolsen Also getting this error when building `[Error] Cannot write tags for C:/Users/crims/Documents/Development/GoA/official/ui-components/libs/web-components/src/components/file-upload-card/FileUploadCard.html-data.json because SyntaxError: Unexpected token } in JSON at position 217`

**Review by ArakTaiRoth — APPROVED** [review]

**Review by vanessatran-ddi — APPROVED** [review]
👍
